In [1]:
from ucimlrepo import fetch_ucirepo 
import pandas as pd
import numpy as np
from sklearn.impute import SimpleImputer
from sklearn.pipeline import Pipeline
from sklearn.metrics import confusion_matrix, accuracy_score
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.model_selection import KFold, cross_val_score, GridSearchCV

In [2]:
# fetch dataset 
breast_cancer_wisconsin_original = fetch_ucirepo(id=15) 
  
# data (as pandas dataframes) 
X = breast_cancer_wisconsin_original.data.features 
y = breast_cancer_wisconsin_original.data.targets 
X.columns

Index(['Clump_thickness', 'Uniformity_of_cell_size',
       'Uniformity_of_cell_shape', 'Marginal_adhesion',
       'Single_epithelial_cell_size', 'Bare_nuclei', 'Bland_chromatin',
       'Normal_nucleoli', 'Mitoses'],
      dtype='object')

In [3]:
y['Class'].value_counts()

Class
2    458
4    241
Name: count, dtype: int64

In [4]:
y['Class'] = np.where(y['Class'] == 2, 0 , 1)
y['Class'].value_counts()

C:\Users\Sanjay Sane\AppData\Local\Temp\ipykernel_29964\2870092414.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  y['Class'] = np.where(y['Class'] == 2, 0 , 1)


Class
0    458
1    241
Name: count, dtype: int64

In [5]:
X.drop('Bare_nuclei', axis=1, inplace=True)
X.isnull().sum()

C:\Users\Sanjay Sane\AppData\Local\Temp\ipykernel_29964\3601396363.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  X.drop('Bare_nuclei', axis=1, inplace=True)


Clump_thickness                0
Uniformity_of_cell_size        0
Uniformity_of_cell_shape       0
Marginal_adhesion              0
Single_epithelial_cell_size    0
Bland_chromatin                0
Normal_nucleoli                0
Mitoses                        0
dtype: int64

In [6]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=24, test_size=0.3)

In [7]:
lr = LogisticRegression()
lr.fit(X_train, y_train)
y_pred = lr.predict(X_test)
#pd.crosstab(y_test['Class'].values, y_pred)

C:\Python\anaconda3\Lib\site-packages\sklearn\utils\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [8]:
print(confusion_matrix(y_test['Class'], y_pred))

[[135   3]
 [  4  68]]


In [9]:
accuracy_score(y_test['Class'], y_pred)

0.9666666666666667

In [10]:
y['Class'].value_counts(normalize=True)

Class
0    0.655222
1    0.344778
Name: proportion, dtype: float64

Naive Rule / Baseline Model

In [12]:
y_pred = np.zeros(210)

In [13]:
y_test['Class'].value_counts()

Class
0    138
1     72
Name: count, dtype: int64

In [14]:
accuracy_score(y_test['Class'], y_pred)

0.6571428571428571

Any ML Model score should be better than score of baseline model.
Here baseline(naive) model score = 0.6571428571428571 & ML Model score = 0.9666666666666667. 
Hence we can say that this ML Model can be a relevant model. 